File: backend/src/models/EmergencyLog.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface IEmergencyLog extends Document {
    userId: string;
    timestamp: Date;
    contactCalled: string;
    contactPhone: string;
    location?: string;
    resolved: boolean;
}

const EmergencyLogSchema: Schema = new Schema({
    userId: { type: String, required: true, index: true },
    timestamp: { type: Date, default: Date.now, required: true },
    contactCalled: { type: String, required: true },
    contactPhone: { type: String, required: true },
    location: { type: String },
    resolved: { type: Boolean, default: false }
}, {
    timestamps: true
});

export default mongoose.model<IEmergencyLog>('EmergencyLog', EmergencyLogSchema);

File: backend-spring/src/main/java/com/aalliswell/entity/EmergencyLog.java
package com.aalliswell.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Index;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.Instant;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "emergency_logs", indexes = @Index(name = "idx_emergency_logs_user_timestamp", columnList = "user_id,timestamp"))
public class EmergencyLog extends AuditableEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private Instant timestamp = Instant.now();

    @Column(nullable = false)
    private String contactCalled;

    @Column(nullable = false)
    private String contactPhone;

    private String location;

    @Column(nullable = false)
    private boolean resolved;
}
